src/main/python/pybuilder/plugins/python/coverage_plugin.py

```diff
@@ -130,38 +142,87 @@ def do_coverage(project, logger, reactor, execution_prefix, execution_name, targ
 
         if module not in modules:
             modules.append(module)
+    return modules
+
+
+def _build_module_report(coverage, module):
+    return ModuleCoverageReport(coverage._analyze(module))
+
+
+def _build_coverage_report(project, logger, execution_name, execution_prefix, coverage, modules):
+    coverage_too_low = False
+    branch_coverage_too_low = False
+    branch_partial_coverage_too_low = False
+    threshold = project.get_property("%s_threshold_warn" % execution_prefix)
+    branch_threshold = project.get_property("%s_branch_threshold_warn" % execution_prefix)
+    branch_partial_threshold = project.get_property("%s_branch_partial_threshold_warn" % execution_prefix)
+
+    report = {
+        "module_names": []
+    }
+
+    sum_lines = 0
+    sum_lines_not_covered = 0
+    sum_branches = 0
+    sum_branches_missing = 0
+    sum_branches_partial = 0
+
+    for module in modules:
+        module_name = module.__name__
 
-        module_report_data = build_module_report(coverage, module)
-        should_ignore_module = module_name in exceptions
+        module_report_data = _build_module_report(coverage, module)
 
-        if not should_ignore_module:
```

Comment:
The idea was that you could exclude a module from coverage (maybe it's legacy code and your other modules are shiny and well-tested) but still look at the reports to look at progress made on adding tests. `continue`ing early has the advantage of being much simpler and less surprising, so I'm positive we should keep this as-is.